✨Enable undo and redo

AdrySky opened the pull request:
# Description

This will enable undo and redo xircuits. Either via context menu or shortcut. Also add shortcut for save.

Shortcut:
1. Ctrl + z = Undo
2. Ctrl + y = Redo
3. Ctrl + s = Save

In addition, change the all the icons to LabIcon and sync it with Jupyterlab theme.

Before in dark theme(Component lib icon): 
![Before dark](https://user-images.githubusercontent.com/84708008/155920444-ef3d342a-7826-4ce9-90bc-4b533d0864df.PNG)


After in dark theme(Component lib icon):
![After dark](https://user-images.githubusercontent.com/84708008/155920375-4731c2af-bb9f-40dc-8ca0-71ab80430bc9.PNG)


## References

Currently, it detect changes mentioned in #75 

## Pull Request Type

- [x] Xircuits Core (Jupyterlab Related changes)
- [x] Xircuits Canvas (Custom RD Related changes)
- [ ] Xircuits Component Library
- [ ] Documentation
- [ ] Others (Please Specify)

## Type of Change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Tests

Based on detected changes mentioned in #75,
1. Check on whether the undo/redo works via shortcuts key.
2. Test also via context menu. 
3. Also, test the icons when changing theme(white -> dark). Previously, the color only works in white theme.

## Tested on?

- [x] Windows  
- [ ] Linux Ubuntu 
- [ ] Centos 
- [ ] Mac  
- [ ] Others  (State here -> xxx )  

# Notes

Probably need to add a listener when nodes position was changed and connected links deleted.

MFA-X-AI reviewed (changes requested):
Thanks for the pull request! Looks like another good QOL feature to have. 
As discussed in our chat, the undo-redo function doesn't work possibly because the detect changes / dirty state isn't functioning properly. Hope ya can fix it. Thanks.

AdrySky commented:
Thanks for the reviewed. I fixed the issue. It related to detecting changes when saving it. I removed it previously because the undo/redo manager detect saving xircuits as changes. Right now, it should be working fine.

MFA-X-AI reviewed (commented):
Great work, the undo and redo is now working!
Was pleasantly surprised when you mentioned that since you've used Juypyter's doc undo and redo there's no limit to the steps, awesome. 

One thing that was odd was that although the undo and redo works perfectly from the right click menu, pressing ctrl z / ctrl y moves backwards / forwards 2 steps. It doesn't break anything else though, so I guess we can merge it if you want and improve it in the future ones.

The changes in the icons are also noted and appreciated (but as always, would prefer if you could make a new pull request for other changes so I can track them easily).
